connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -307,15 +307,21 @@ private void processConnectorConfigUpdates(Set<String> connectorConfigUpdates) {
     }
 
     private void processTargetStateChanges(Set<String> connectorTargetStateChanges) {
-        if (!connectorTargetStateChanges.isEmpty()) {
-            for (String connector : connectorTargetStateChanges) {
-                if (worker.connectorNames().contains(connector)) {
-                    TargetState targetState = configState.targetState(connector);
-                    worker.setTargetState(connector, targetState);
-                    if (targetState == TargetState.STARTED)
-                        reconfigureConnectorTasksWithRetry(connector);
-                }
+        for (String connector : connectorTargetStateChanges) {
+            TargetState targetState = configState.targetState(connector);
```

Comment:
I'm realizing we can actually get into some other weird states here too. I think we might need to check that this doesn't come back `null`. Because we simply accumulate the set of connectors that had their target state change, it's possibly the target state has been deleted once we actually process this.

Not sure if we want to wrap that change into this set of changes as well -- it's getting beyond the scope of what was intended here. It's also probably not common (only occurs if operations get coalesced into a single call to this method). But we might want to file a follow up JIRA; I'd imagine we have similar issues with `processConnectorConfigUpdates`.